### DISCUSSION TITLE
Free Memory Pointer

### DISCUSSION THREAD

**Discussion by Kenn-eth** [Cyfrin/assembly-evm-opcodes-and-formal-verification-course#18] [Q&A] [answered]
I understand that 0x40(64) is reserved in solidity as free memory pointer. That means from 0x60(96) downwards are the free memory. In lesson 43 why did we skip 0x60(96) to point at 0x80(128)?

**Comment by EngrPips** [comment]
Hello @Kenn-eth, This is a very good question I haven't been able to answer myself. I mean, we know `0x00` to `0x40` is taken up for some purpose, but `0x60` upward is free to use, but we started using from `0x80` instead of `0x60.` Let's look forward to @PatrickAlphaC  response when they are available

**Comment by PatrickAlphaC** [comment, marked as answer]
Great question. The [solidity docs](https://docs.soliditylang.org/en/latest/internals/layout_in_memory.html) actually do a great job explaining this.

Essentially:
0x00 -> 0x40: Scratch space for hashing functions
0x40 -> 0x60: Free Memory Pointer
0x60 -> 0x80: zero slot

According to the documentation:

> The zero slot is used as initial value for dynamic memory arrays and should never be written to (the free memory pointer points to 0x80 initially).

So that's why!

**Reply by EngrPips** [reply]
Interesting

**Reply by Kenn-eth** [reply]
Thank you Patrick.
